tests/ludwig/datasets/titanic/test_titanic_workflow.py

```diff
@@ -0,0 +1,72 @@
+import tempfile
+import pandas as pd
+from unittest import mock
+from ludwig.datasets.base_dataset import DEFAULT_CACHE_LOCATION
+from ludwig.datasets.titanic import Titanic
+
+
+class FakeTitanicDataset(Titanic):
+    def __init__(self, cache_dir=DEFAULT_CACHE_LOCATION):
+        super().__init__(cache_dir=cache_dir)
+
+
+def test_download_titanic_dataset():
+    titanic_train_df = pd.DataFrame({
+        'passenger_id': [1216, 699, 234],
+        'pclass': [3, 3, 4],
+        'name': ['sai bo', 'bo staff', 'tae kwan nic'],
+        'sex': ["female", "male", "male"],
+        'age': [38, 28, 18],
+        'sibsp': [0,1,0],
+        'parch': [1,1,2],
+        'ticket': [335432, 315089, 322472],
+        'fare': [7.7333, 8.6625, 9.8765],
+        'cabin': [1, 2, 4],
+        'embarked': ["C", "Q", "S"],
+        'boat': [0, 0, 0],
+        'body': [0, 1, 0],
+        'home.dest': ["Croatia", "Italy","Sweden"],
+        'survived': [0, 1, 0]
+    })
+
+    titanic_test_df = pd.DataFrame({
+        'passenger_id': [1216, 699, 234],
+        'pclass': [3, 3, 4],
+        'name': ['mo bo', 'bo bo bo', 'Rafael Nadal'],
+        'sex': ["female", "male", "male"],
+        'age': [28, 18, 30],
+        'sibsp': [0, 1, 0],
+        'parch': [1, 1, 2],
+        'ticket': [335412, 215089, 922472],
+        'fare': [17.7333, 18.6625, 19.8765],
+        'cabin': [2, 2, 1],
+        'embarked': ["Q", "Q", "C"],
+        'boat': [0, 0, 0],
+        'body': [0, 1, 0],
+        'home.dest': ["Sweden", "Slovenia", "Italy"],
+        'survived': [0, 1, 0]
+    })
+    titanic_train_filename = "titanic_train.csv"
+    titanic_test_filename = "titanic_test.csv"
+    with tempfile.TemporaryDirectory() as source_dir:
```

Comment:
In discussions with @w4nderlust about the unit tests in general, we looking to standardize on [pytest's fixture `tmpdir`](https://docs.pytest.org/en/stable/tmpdir.html#the-tmpdir-fixture) for temporary directories.  We're in the process of updating the other tests to use this fixture.

Is it possible for this test to be adapted to use pytest's `tmpdir` fixture?

I see later on that this test creates a second temporary directory within the first temporary directory (`source_dir`).  This comment does not apply to the second temporary directory.  I think it can be your call on how you manage internal temporary directories.  About the only consideration is the potential naming conflict between your second temporary directory and pytest's `tmpdir` fixture.

Reply:
sure will change @jimthompson5802 , thanks for the input and moving forward please flag me if appropriate for some of those discussions